File: csharp/basics/Fibonacci.cs
// Print the first 20 Fibonacci numbers from an iterator method built on yield return.
// Run: dotnet run Fibonacci.cs (file-based app, .NET 10 or later), or as Program.cs in a console project

Console.WriteLine(string.Join(" ", Fibonacci().Take(20)));

static IEnumerable<long> Fibonacci()
{
    (long current, long next) = (0L, 1L);
    while (true)
    {
        yield return current;
        (current, next) = (next, current + next);
    }
}

File: clojure/basics/fibonacci.clj
;; Print the first 20 Fibonacci numbers from a lazy sequence built with iterate.
;; Run: clojure -M fibonacci.clj

(require '[clojure.string :as string])

(def fibonacci
  (map first (iterate (fn [[current next]] [next (+ current next)]) [0 1])))

(println (string/join " " (take 20 fibonacci)))
